recipes/binutils/all/conanfile.py

```diff
@@ -1,70 +1,154 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import functools
 import os
-import re
-from glob import glob
 
-from conan import ConanFile
-from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
-from conans import tools
-from conans.errors import ConanInvalidConfiguration
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.40.0"
 
 
 class BinutilsConan(ConanFile):
     name = "binutils"
-    description = "a collection of binary tools"
-    topics = ("gnu", "compilation", "tools")
-    url = "https://github.com/conan-io/conan-center-index"
-    homepage = "https://www.gnu.org/software/binutils"
+    description = "The GNU Binutils are a collection of binary tools."
     license = "GPL-2.0-or-later"
-
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://www.gnu.org/software/binutils"
+    topics = ("binutils", "ld", "linker", "as", "assembler", "objcopy", "objdump")
     settings = "os", "arch", "compiler", "build_type"
 
-    _source_subfolder = "source_subfolder"
+    options = {
+        "multilib": [True, False],
+        "with_libquadmath": [True, False],
+        "target_arch": "ANY",
+        "target_os": "ANY"
+    }
+
+    default_options = {
+        "multilib": True,
+        "with_libquadmath": True,
+        "target_arch": None,  # Initialized in config_options
+        "target_os": None,  # Initialized in config_options
+    }
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+        self.options.target_arch = str(getattr(self, "settings_target", self.settings).arch)
+        self.options.target_os = str(getattr(self, "settings_target", self.settings).os)
 
     def validate(self):
-        if self.settings.os not in ["Linux", "FreeBSD"]:
+        if self.settings.os not in ("Linux", "FreeBSD"):
```

Comment:
@hlewin

I've kept this validation.
This line means the cci recipe currently supports building binutils on Linux/FreeBSD.

But the recipe allows you to build binutils that targets Windows/FreeBSD/Macos/...

In the test_package, some architectures are tested.
I've verified that the executables targetting: x86@Linux, x86_64@Linux, x86@Windows and x86_64@Windows work (using wine)
I did not verify the armv8@Linux executable (yet)